src/Microsoft.AspNetCore.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -18,17 +18,37 @@ public class ObjectAdapter : IObjectAdapterWithTest
         /// <param name="logErrorAction">The <see cref="Action"/> for logging <see cref="JsonPatchError"/>.</param>
         public ObjectAdapter(
             IContractResolver contractResolver,
-            Action<JsonPatchError> logErrorAction)
+            Action<JsonPatchError> logErrorAction):
+            this(contractResolver, logErrorAction, new AdapterFactory())
         {
-            ContractResolver = contractResolver ?? throw new ArgumentNullException(nameof(contractResolver));
-            LogErrorAction = logErrorAction;
         }
 
+        /// <summary>
+         /// Initializes a new instance of <see cref="ObjectAdapter"/>.
+         /// </summary>
+         /// <param name="contractResolver">The <see cref="IContractResolver"/>.</param>
+         /// <param name="logErrorAction">The <see cref="Action"/> for logging <see cref="JsonPatchError"/>.</param>
+        /// <param name="adapterFactory">The <see cref="IAdapterFactory"/> to use when creating adaptors.</param>
+         public ObjectAdapter(
+            IContractResolver contractResolver,
+            Action<JsonPatchError> logErrorAction,
+            IAdapterFactory adapterFactory)
+         {
+             ContractResolver = contractResolver ?? throw new ArgumentNullException(nameof(contractResolver));
+            LogErrorAction = logErrorAction;
+            AdapterFactory = adapterFactory ?? throw new ArgumentNullException(nameof(adapterFactory));
+         }
+
         /// <summary>
         /// Gets or sets the <see cref="IContractResolver"/>.
         /// </summary>
         public IContractResolver ContractResolver { get; }
 
+        /// <summary>
+        /// Gets or sets the <see cref="IAdapterFactory"/>
+        /// </summary>
+        public IAdapterFactory AdapterFactory { get; set; }
```

Comment:
Does this need to be publicly settable since it's passed in the constructor?

Reply:
@HappyNomad and I had discussed this earlier.  I tend to leave them settable for flexibility and habits kicked in. Removed.